provider/machine_pool_resource.go

```diff
@@ -247,6 +251,59 @@ func (r *MachinePoolResource) Create(ctx context.Context,
 		return
 	}
 
+	// The default machine pool is created automatically when the cluster is created.
+	// We want to import it instead of creating it.
+	if machinepoolName == defaultMachinePoolName {
```

Comment:
I think it will be more clear to move it to another method

Reply:
done